tasty-plutus/src/Test/Tasty/Plutus/Internal/Context.hs

```diff
@@ -68,17 +68,17 @@ import Prelude hiding (length)
 
  @since 1.0
 -}
-data Purpose
-  = -- | Corresponds to 'Plutus.V1.Ledger.Contexts.Minting'. This tag applies
-    -- to minting-policy-related structures and functions.
-    --
-    -- @since 1.0
-    ForMinting
-  | -- | Corresponds to 'Plutus.V1.Ledger.Contexts.Spending'. This tag applies
-    -- to validator-related structures and functions.
-    --
-    -- @since 1.0
-    ForSpending
+data Purpose where
+  -- | Corresponds to 'Plutus.V1.Ledger.Contexts.Minting'. This tag applies
+  -- to minting-policy-related structures and functions.
+  --
+  -- @since 1.0
+  ForMinting :: forall redeemer. redeemer -> Purpose
+  -- | Corresponds to 'Plutus.V1.Ledger.Contexts.Spending'. This tag applies
+  -- to validator-related structures and functions.
+  --
```

Comment:
I think we should specify the version in the changes:
```haskell
-- | @since 1.0
ForMinting ::
  forall (redeemer :: Type).
  -- | @since 5.1
  redeemer ->
  Purpose
```